packages/babel-helpers/src/index.js

```diff
@@ -62,15 +80,19 @@ function getHelperMetadata(file) {
 
       Object.keys(bindings).forEach(name => {
         if (name === exportName) return;
+        if (dependencies.has(bindings[name].identifier)) return;
 
         localBindingNames.add(name);
       });
     },
     ReferencedIdentifier(child) {
       const name = child.node.name;
-      const binding = child.scope.getBinding(name);
-
-      if (!binding) globals.add(name);
+      const binding = child.scope.getBinding(name, /* noGlobal */ true);
+      if (!binding) {
+        globals.add(name);
+      } else if (dependencies.has(binding.identifier)) {
+        importBindingsReferences.push(makePath(child));
```

Comment:
What's the motivation for putting this here instead of doing it above where you have `dependencies.set(bindingIdentifier, name)`?

Reply:
When I do `dependencies.set(bindingIdentifier, name)` I don't have access to the NodePath of the reference, I can only use the path of the identifier inside the import statement.

Reply:
Ahh of course, I just missed that they were different nodes.